D8CORE-6824: Spacer paragraph - changed label of default value

imonroe opened the pull request:
# READY FOR REVIEW

# Summary
- We want to change the label of the `- None -` option on the Spacer paragraph to be `Standard`.

# Review By (Date)
- code freeze

# Criticality
- 2

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Check out the [corresponding branch in Stanford Profile](https://github.com/SU-SWS/stanford_profile/pull/700) to see the updated help text in the UI
3. Rebuild Cache and import config `drush cr ; drush ci`
4. Navigate to any Layout Builder page.
5. Create a Spacer paragraph wherever is convenient
6. Verify the default size option of the spacer is now `Standard`

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? No, but the help text is updated in Stanford Profile, and that one does

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

imonroe committed:
D8CORE-6824: Spacer paragraph - changed label of default value

imonroe committed:
codeclimate fixup

pookmish commented on modules/stanford_layout_paragraphs/stanford_layout_paragraphs.module:
```diff
@@ -95,3 +95,13 @@ function _stanford_layout_paragraphs_is_editing_layout_paragraphs() {
+function stanford_layout_paragraphs_form_layout_paragraphs_component_form_alter(&$form, &$form_state) {
+  // Modify the labels on the Spacer paragraph select list.
+  if (!empty($form['su_spacer_size'])) {
+    $form['su_spacer_size']['widget']['#options']['_none'] = 'Standard';
```
this would be better suited in a widget form alter:
https://github.com/SU-SWS/stanford_profile_helper/blob/9.x/src/EventSubscriber/FormEventSubscriber.php#L48

imonroe replied:
Ok by me.  I'll update it.

imonroe committed:
Moved to FormEventSubscriber

imonroe committed:
codeclimate fixup

imonroe committed:
added test

tests

tests

tests

tests

testing tests

tests

tests

test

tests

tests

tests

tests

test

test

tests

tests

test

test

test

test

test

test

test

test

test

test

added a cache clear

test

test

imonroe committed:
removed debugging code

pookmish committed:
Refactored test

fixup - test

revert previous change

test

test

test

test

test

test

test

tests

test

test with custom mock form

test

test

test

test

test

updated event name

use FieldHookEvents

more context

test

test

debug

debug

debug

test

almost there

test

with debugging

test

frustrating.

test

test

test

test

test

test

test

test

imonroe committed:
fixup

imonroe commented:
finally got the test working. Six lines of code to change the label, 120 lines of code for the test.  Is there a way I could have done this more efficiently?
